### ISSUE TITLE
lmp in 2.1.3 make unrecognized pair error

### ISSUE THREAD

**Issue by pee8379** [deepmodeling/deepmd-kit#1809]
### Bug summary

I try to continue training model (from deepmd-kit 2.1.1, dpgen 0.10.5), using newly released 2.1.3. (offline package deepmd-kit-2.1.3-cuda10.2_gpu-Linux-x86_64.sh )

dpgen process failed to run, (task 4) with [unrecognized pair style] error in lmp log file.

### DeePMD-kit Version

2.1.3

### TensorFlow Version

-

### How did you download the software?

Offline packages

### Input Files, Running Commands, Error Log, etc.

dpgen 0.10.5
call lmp by 
"model_devi":[ "command": "/DeepMD-kit-location/deepmd-kit-2.1.1-cu10.2/bin/lmp"
in machine json file.

### Steps to Reproduce

-

**Comment by njzjz** [comment]
Please [activate](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#activating-an-environment) the environment first:
```sh
conda activate /path/to/deepmd-kit
```

（`source activate` may also work)

**Comment by r-hou** [comment]
it helps nothing, I did activate the environment, otherwise how could I use "lmp -h", it doesn't show deepmd pair_style

**Comment by njzjz** [comment]
> it helps nothing, I did activate the environment, otherwise how could I use "lmp -h", it doesn't show deepmd pair_style

Currently, `lmp -h` only lists built-in styles, but no plugged styles. This part of the codes is maintained by LAMMPS.

**Comment by njzjz** [comment]
> I have activated the environment but I still get the error message: "ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:271)" Do you have any suggestions?

If it was installed from conda-forge or the offline package released recently, the problem has been resolved in https://github.com/conda-forge/deepmd-kit-feedstock/pull/63.

----

During the activation process, the environment variable `LAMMPS_PLUGIN_PATH` will be set, so it can also be manually set instead:

```sh
export LAMMPS_PLUGIN_PATH=$CONDA_PREFIX/lib/deepmd_lmp
```

**Comment by hyuqing11** [comment]
Thank you for your help. However, I encountered an issue after setting the LAMMPS_PLUGIN_PATH. The error message I received is as follows: "Open of file /test_env4/lib/deepmd_lmp/dpplugin.so failed: libabsl_status.so.2301.0.0: cannot open shared object file: No such file or directory."  Do you have any suggestions?

**Comment by chazeon** [comment]
@njzjz @hyuqing11

I have encounter the same kind of problem today.
The `deepmd_lmp/dpplugin.so` from the current conda-forge version of deepmd-kit version does not has proper rpath and needed shared library information.

After installation, a temporary patch can be done with `patchelf` with
```sh
cd $CONDA_PREFIX/lib
patchelf --add-rpath '$ORIGIN/..' deepmd_lmp/dpplugin.so
patchelf --add-needed liblammps.so deepmd_lmp/dpplugin.so
```

My deepmd-kit version and build is
```
deepmd-kit                2.2.7           cuda118py310h49a02c8_3    conda-forge
```

**Comment by njzjz** [comment]
> The `deepmd_lmp/dpplugin.so` from the current conda-forge version of deepmd-kit does not has proper rpath and needed shared library information.

I know @Yi-FanLi had this issue before, but he compiled LAMMPS himself. @chazeon Did you use conda-forge's LAMMPS or others?

**Comment by chazeon** [comment]
@chazeon Yes, I used conda-forge’s LAMMPS.

**Comment by chazeon** [comment]
> I know @Yi-FanLi had this issue before, but he compiled LAMMPS himself. @chazeon Did you use conda-forge's LAMMPS or others?

Yes, I used conda-forge’s LAMMPS.

**Comment by njzjz** [comment]
> Yes, I used conda-forge’s LAMMPS.

Do you use `lmp_mpi` or `lmp_serial`? `lmp_serial` in conda-forge does not provide a shared library (`liblammps.so` is built for MPI), which might be a problem and should not be used.

**Comment by chazeon** [comment]
@njzjz @Yi-FanLi
I see. Currently, `lmp_mpi` indeed loads `liblammps.so` automatically and then loads the plugin correctly. I was fixing it because `ldd` or `readelf` does not list `liblammps.so` as a dependency.

It's my fault. I spotted the missing `.so` because an earlier incompatible version of LAMMPS was installed by `conda` because of other dependencies pinned down the LAMMPS version. I tried to debug the `.so` file to look into its symbols. I wasn't aware that upgrading LAMMPS version has already fixed these issues.
